Implementation of nautobot-server startplugin. (#404)
nautobot/nautobot: nautobot/docs/administration/nautobot-server.md

Please see the guide on [Deploying Nautobot](../installation/wsgi.md) for our recommended configuration for use with uWSGI.

### `startplugin`

`nautobot-server startplugin <name> [directory]`

Create a new plugin with `name`.

This command is similar to the django-admin [startapp](https://docs.djangoproject.com/en/dev/ref/django-admin/#startapp) command, but with a default template directory of `nautobot/core/template/plugin_template`. This command assists with creating a basic file structure for beginning development of a new Nautobot plugin.

Without passing in the destination directory, `nautobot-server startplugin` will use your current directory and the `name` argument provided to create a new directory. We recommend providing a directory so that the plugin can be installed or published easily. Here is an example:

> 

Reviewer: In keeping with the other commands, can you please add documentation for the optional arguments here?

e.g. `--name`, `--extension`, `--template`, etc.

Author: I started to, but these are all coming from startapp, and better documented there than I could do justice here.  If you still think it should belong here, then I will add it.

```

Please see the guide on [Troubleshooting Webhooks](../models/extras/webhook.md#troubleshooting) for more information.

> ### `startplugin`

Reviewer 2: Looks like this file lists commands alphabetically, so please move this section up as appropriate.

> `nautobot-server startplugin [app_name]`

Reviewer 2: Looks like this file uses `[]` interchangeably to denote both optional arguments and variable arguments. Personally I'd rather be clearer here:

```suggestion
`nautobot-server startplugin <plugin_name> [<directory>]`
```

Create a new plugin with `app_name`.

> This command is just like the django-admin [startapp](https://docs.djangoproject.com/en/dev/ref/django-admin/#startapp). But has the template directory set to `nautobot/core/template/plugin_template`. This command assists with creating the Nautobot plugin folder structure.

Reviewer 2: ```suggestion
This command is similar to the django-admin [startapp](https://docs.djangoproject.com/en/dev/ref/django-admin/#startapp) command, but with a default template directory of `nautobot/core/templates/plugin_template/`. This command assists with creating a basic file structure for beginning development of a new Nautobot plugin.
```

> 

Reviewer 2: It'd be nice to provide an example either here or in the plugin dev docs, as it's not immediately obvious to me where the created plugin would be by default if not specifying the `directory` parameter, nor whether I should be specifying the project root directory (i.e. `~/my-plugin/` or the Python package root directory (`~/my-plugin/my_plugin/`) as a parameter.

Author: Sure, it defaults to whatever directory you issue the command from (which could be anywhere) and by default creates a directory with whatever the `name` input was.  So, in your example `<cwd>/my-plugin/`.

> Additional options can be found in the django [documentation](https://docs.djangoproject.com/en/dev/ref/django-admin/#startapp).

Reviewer 2: ```suggestion
Additional options can be found in the Django [documentation](https://docs.djangoproject.com/en/dev/ref/django-admin/#startapp).
```
